File: python/difference-of-squares/difference_of_squares.py

def square_of_sum(number):
    number_sum = sum([i for i in range(1, number+1)])
    return number_sum ** 2


def sum_of_squares(number):
    return sum([i ** 2 for i in range(1, number+1)])


def difference_of_squares(number):
    return square_of_sum(number) - sum_of_squares(number)

File: ruby/difference-of-squares/difference_of_squares.rb
class Squares
  def initialize(args)
    @num = args
  end

  def square_of_sum
    (1..@num).to_a.reduce(:+)**2
  end

  def sum_of_squares
     (1..@num).reduce(0) { |sum, e| sum + e*e }
  end

  def difference
    square_of_sum - sum_of_squares
  end
end

module BookKeeping
    VERSION = 4
end